### ISSUE TITLE
permissions.name too short if using wildcard permissions with UUIDs

### ISSUE THREAD

**Issue by Cluster2a** [spatie/laravel-permission#2431]
**Describe the bug**
When working with UUIDs and wildcard permissions (shiro-strings), the permission string gets pretty long:
`bookings.viewAny,view,create,update,deleteOwn,addStatus,updateStatus.<UUID_1>,<UUID_2>,<UUID_3>,<UUID_4>`

As the field `permissions.name` is only 255 chars long (`varchar(255)`) the strings are cut off.

Since the name is part of the index I can't just alter the name field to a text field - I probably have to remove the index and take care of duplicate records via logic? As an alternative, I could add a field that holds the hash of the `permissions.name` and add a unique index on the `name_hash` and `guard_name`.

Are there any suggestions?

**Versions**
You can use `composer show` to get the version numbers of:
- spatie/laravel-permission package version: 5.10.1
- illuminate/framework package: v9.52.7

PHP version: 8.2

Database version: 8.0.33

**To Reproduce**
- use wildcard permissions with multiple UUIDs longer than 255 chars

**Comment by parallels999** [comment]
the hash thing seems like a good idea, 
fill the hash on `saving` or `updating`

**Comment by drbyte** [comment]
> I could add a field that holds the hash of the permissions.name and add a unique index on the name_hash and guard_name.

Yes, that is probably the simplest solution.

**Comment by Cluster2a** [comment]
> > I could add a field that holds the hash of the permissions.name and add a unique index on the name_hash and guard_name.
> 
> Yes, that is probably the simplest solution.

Should I close this issue knowing that this is more of a workaround?

**Comment by drbyte** [comment]
> Should I close this issue knowing that this is more of a workaround?

I'd be curious to hear back how you implemented it. Might be worth incorporating ...

**Comment by Cluster2a** [comment]
> > Should I close this issue knowing that this is more of a workaround?
> 
> I'd be curious to hear back how you implemented it. Might be worth incorporating ...

I added the hash this way:

```php
class Permission extends SpatiePermission
{
    use Uuid;

    /**
     * @return void
     */
    protected static function booted(): void
    {
        static::creating(function (Permission $permission) {
            $permission->name_hash = hash('xxh128', $permission->name);
        });

        static::updating(function (Permission $permission) {
            $permission->name_hash = hash('xxh128', $permission->name);
        });
    }
}
```

**Comment by drbyte** [comment]
Yes, that makes sense.

I would add a note to anyone who is (unlikely) exploring this on older PHP versions, take note that `xxh128` is only available since PHP 8.1 ... if you're trying to do this on PHP 7.4 or 8.0 then you will probably need to use `crc32c`.
